internal/common/login_test.go

```diff
@@ -0,0 +1,95 @@
+package common
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/metal-toolbox/auditevent"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestRemoteUserLogin_Validate(t *testing.T) {
+	t.Parallel()
+
+	rul := RemoteUserLogin{
+		Source:     &auditevent.AuditEvent{},
+		PID:        666,
+		CredUserID: "foo",
+	}
+
+	err := rul.Validate()
+
+	assert.Nil(t, err)
+}
+
+func TestRemoteUserLogin_Validate_SourceIsNil(t *testing.T) {
+	t.Parallel()
+
+	rul := RemoteUserLogin{
+		Source:     nil,
+		PID:        666,
+		CredUserID: "foo",
+	}
+
+	err := rul.Validate()
+	var exp *RemoteUserLoginValidateError
+	if !errors.As(err, &exp) {
```

Comment:
you could use `assert.ErrorAs(...)` instead [1].

If you really need the `Fatalf` call, you could do:

```
require.ErrorAs(t, err, &exp, "expected %T - got %T", exp, err)
```

`require` does a similar thing as `Fatal` and also comes from the testify assertion library.

[1] https://pkg.go.dev/github.com/stretchr/testify/assert#ErrorAs

Reply:
It's frustrating that this library not only exists but also uses the word `assert` to do the opposite of what `assert` does in C.

Reply:
Change has been made.